node/src/main/scala/coop/rchain/node/web/package.scala

```diff
@@ -24,59 +24,58 @@ package object web {
   def corsPolicy[F[_]: Sync](routes: HttpRoutes[F]) =
     CORS(routes, CORS.DefaultCORSConfig.copy(allowCredentials = false))
 
-  def aquireHttpServer[F[_]: ConcurrentEffect: Timer: RPConfAsk: NodeDiscovery: ConnectionsCell: EventConsumer: Log](
+  def acquireHttpServer[F[_]: ConcurrentEffect: Timer: RPConfAsk: NodeDiscovery: ConnectionsCell: Log](
       reporting: Boolean,
       host: String = "0.0.0.0",
       httpPort: Int,
       prometheusReporter: NewPrometheusReporter,
       connectionIdleTimeout: FiniteDuration,
       webApi: WebApi[F],
       reportingRoutes: ReportingHttpRoutes[F]
-  )(implicit scheduler: Scheduler): F[fs2.Stream[F, ExitCode]] =
-    for {
-      event              <- EventsInfo.service[F]
-      reportingRoutesOpt = if (reporting) reportingRoutes else HttpRoutes.empty
-      baseRoutes = Map(
-        "/metrics"   -> corsPolicy(NewPrometheusReporter.service[F](prometheusReporter)),
-        "/version"   -> corsPolicy(VersionInfo.service[F]),
-        "/status"    -> corsPolicy(StatusInfo.service[F]),
-        "/ws/events" -> corsPolicy(event),
-        "/api"       -> corsPolicy(WebApiRoutes.service[F](webApi) <+> reportingRoutesOpt),
-        // Web API v1 with OpenAPI schema
-        "/api/v1" -> corsPolicy(WebApiRoutesV1.create[F](webApi))
-      )
-      // Legacy reporting routes
-      extraRoutes = if (reporting)
+  )(implicit scheduler: Scheduler): Resource[F, Server[F]] = {
+    val reportingRoutesOpt = if (reporting) reportingRoutes else HttpRoutes.empty
+    val baseRoutes = Map(
+      "/metrics" -> corsPolicy(NewPrometheusReporter.service[F](prometheusReporter)),
+      "/version" -> corsPolicy(VersionInfo.service[F]),
+      "/status"  -> corsPolicy(StatusInfo.service[F]),
+      "/api"     -> corsPolicy(WebApiRoutes.service[F](webApi) <+> reportingRoutesOpt),
+      // Web API v1 with OpenAPI schema
+      "/api/v1" -> corsPolicy(WebApiRoutesV1.create[F](webApi))
+    )
+    // Legacy reporting routes
+    val extraRoutes =
+      if (reporting)
         Map("/reporting" -> corsPolicy(reportingRoutes))
       else
         Map.empty
-      allRoutes = baseRoutes ++ extraRoutes
-    } yield BlazeServerBuilder[F](scheduler)
+    val allRoutes = baseRoutes ++ extraRoutes
+
+    BlazeServerBuilder[F](scheduler)
       .bindHttp(httpPort, host)
       .withHttpApp(RouterFix(allRoutes.toList: _*).orNotFound)
       .withIdleTimeout(connectionIdleTimeout)
       .withResponseHeaderTimeout(connectionIdleTimeout - 1.second)
-      .serve
+      .resource
```

Comment:
Many libraries are exposed as Resource and can be easily combined and safely released after use.